### ISSUE TITLE
`Auth.VerifyForUser` returns response data as error

### ISSUE THREAD

**Issue by cezarsmpio** [supabase-community/auth-go#2]
# Bug report

## Describe the bug

When calling `Auth.VerifyForUser` I expect the first parameter, which is the response, to be filled, instead, it's `nil`.

Example: 

```go
	res, err := s.supabaseService.Client().Auth.VerifyForUser(types.VerifyForUserRequest{
		Type:       types.VerificationType("email"),
		Token:      authSignUpConfirmDto.Code,
		Email:      authSignUpConfirmDto.Email,
		RedirectTo: siteUrl,
	})

	fmt.Println("res", res)
	fmt.Println("err", err)
```

If this succeeds, `res` is `nil`, and `err` contains the response with access token and other data.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Call `Auth.VerifyForUser` with correct data
2. First parameter, which is type of `types.VerifyForUserRequest` is `nil`, which it shouldn't if the response succeeds

## Expected behavior

I expect that `types.VerifyForUserRequest` is not `nil`, and contains access token.

I assume, if this is indeed a bug, the error might be between these lines: https://github.com/supabase-community/auth-go/blob/main/endpoints/verify.go#L142-L148

Maybe Supabase API doesn't return `303` anymore?

## Screenshots

`res` is `nil`

![image](https://github.com/user-attachments/assets/26c1b7d0-c9e5-4b58-951b-f4b190437f5e)

## System information

- OS: macOS
- gotrue-go: `1.2.0`
- supabase-go: `0.0.4`

**Comment by kwoodhouse93** [comment]
Should be addressed by #8

**Comment by cezarsmpio** [comment]
Thanks 😊 you guys rock!
